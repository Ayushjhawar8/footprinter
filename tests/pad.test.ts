import { expect, test } from "bun:test"
import { fp } from "../src/footprinter"

test("pad footprint", () => {
  const soup = fp().pad().w(2).h(1).circuitJson()
  expect(soup).toMatchSnapshot()
})

test("pad footprint with different dimensions", () => {
  const soup = fp().pad().w(3).h(2).circuitJson()
  expect(soup).toMatchSnapshot()
})
