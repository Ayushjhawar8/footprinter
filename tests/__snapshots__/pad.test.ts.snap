// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`pad footprint 1`] = `
[
  {
    "height": 1,
    "layer": "top",
    "pcb_smtpad_id": "",
    "port_hints": [
      "1",
    ],
    "shape": "rect",
    "type": "pcb_smtpad",
    "width": 2,
    "x": 0,
    "y": 0,
  },
  {
    "anchor_alignment": "center",
    "anchor_position": {
      "x": 0,
      "y": 1,
    },
    "font": "tscircuit2024",
    "font_size": 0.2,
    "layer": "top",
    "pcb_component_id": "pcb_component_1",
    "pcb_silkscreen_text_id": "silkscreen_text_1",
    "text": "{REF}",
    "type": "pcb_silkscreen_text",
  },
]
`;

exports[`pad footprint with different dimensions 1`] = `
[
  {
    "height": 2,
    "layer": "top",
    "pcb_smtpad_id": "",
    "port_hints": [
      "1",
    ],
    "shape": "rect",
    "type": "pcb_smtpad",
    "width": 3,
    "x": 0,
    "y": 0,
  },
  {
    "anchor_alignment": "center",
    "anchor_position": {
      "x": 0,
      "y": 1.5,
    },
    "font": "tscircuit2024",
    "font_size": 0.2,
    "layer": "top",
    "pcb_component_id": "pcb_component_1",
    "pcb_silkscreen_text_id": "silkscreen_text_1",
    "text": "{REF}",
    "type": "pcb_silkscreen_text",
  },
]
`;
